pennylane_sf/remote.py

```diff
@@ -0,0 +1,152 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Strawberry Fields Hardware device
+=================================
+
+**Module name:** :mod:`pennylane_sf.hw`
+
+.. currentmodule:: pennylane_sf.hw
+
+The Strawberry Fields Hardware plugin implements all the :class:`~pennylane.device.Device` methods,
+and provides access to Xanadu's continuous-variable quantum hardware.
+
+Classes
+-------
+
+.. autosummary::
+   StrawberryFieldsRemote
+
+
+Code details
+~~~~~~~~~~~~
+"""
+
+from collections import OrderedDict
+
+import numpy as np
+
+import strawberryfields as sf
+# import state preparations, gates and measurements
+from strawberryfields.ops import (BSgate, Catstate, CKgate, Coherent, CXgate,
+                                  CZgate, DensityMatrix, Dgate,
+                                  DisplacedSqueezed, Fock, Gaussian,
+                                  Interferometer, Ket, Kgate, MeasureFock,
+                                  Pgate, Rgate, S2gate, Sgate, Squeezed,
+                                  Thermal, Vgate)
+from strawberryfields.utils.post_processing import (all_fock_probs_pnr,
+                                                    samples_expectation,
+                                                    samples_variance)
+
+from .simulator import StrawberryFieldsSimulator
+
+
+class StrawberryFieldsRemote(StrawberryFieldsSimulator):
+    r"""StrawberryFields remote device for PennyLane.
+
+    A valid Strawberry Fields API token is needed for access. This token can be
+    passed when creating the device. The default configuration options of
+    Strawberry Fields are used to store the authentication token in a
+    configuration file.
+
+    Args:
+        wires (int): the number of modes to initialize the device in
+        shots (int): number of circuit evaluations/random samples used to
+            estimate expectation values of observables
+        backend (str): name of the remote backend to be used
+        hbar (float): the convention chosen in the canonical commutation
+            relation :math:`[x, p] = i \hbar`
+        sf_token (str): the SF API token used for remote access
+    """
+    name = "Strawberry Fields Hardware PennyLane plugin"
+    short_name = "strawberryfields.ai"
```

Comment:
@josh146: should this also be changed to `"strawberryfields.remote"` :thinking:

Reply:
I think so! Other options are `strawberryfields.cloud`, + maybe others?

Reply:
Thanks! Went with `"strawberryfields.remote"`